src/serializers/fields.rs

```diff
@@ -72,14 +75,30 @@ impl SerField {
     }
 }
 
-fn exclude_default(value: &Bound<'_, PyAny>, extra: &Extra, serializer: &CombinedSerializer) -> PyResult<bool> {
+fn exclude_default_or_if(
+    exclude_if_callable: &Option<Py<PyAny>>,
+    value: &Bound<'_, PyAny>,
+    extra: &Extra,
+    serializer: &CombinedSerializer,
+) -> PyResult<bool> {
+    let py = value.py();
+
```

Comment:
think this should be called only if any condition is met, right? inside the `if`

Reply:
I think we should probably handle this outside of the `exclude_default` function as to decouple them for readability. We don't couple `exclude_default` with `exclude_none` or `exclude_unset`, so I think we should retain separation here as well!